packages/date-picker/test/has-input-value.test.js

```diff
@@ -0,0 +1,84 @@
+import { expect } from '@esm-bundle/chai';
+import { fixtureSync } from '@vaadin/testing-helpers';
+import { sendKeys } from '@web/test-runner-commands';
+import sinon from 'sinon';
+import '../src/vaadin-date-picker.js';
+import { waitForScrollToFinish } from './helpers.js';
+
+describe('has-input-value-changed event', () => {
```

Comment:
Consider creating a separate test suite e.g. `events.test.js` so that we could move some tests there:


https://github.com/vaadin/web-components/blob/66bc6b5287d0143ecf043a665a4615cf825f7d82/packages/date-picker/test/keyboard-input.test.js#L576

https://github.com/vaadin/web-components/blob/66bc6b5287d0143ecf043a665a4615cf825f7d82/packages/date-picker/test/keyboard-input.test.js#L739

It would be great to do that in a separate PR and then backport it at least to 24.0 and maybe 23.3 too.

Reply:
I was actually planning to extract `change and validate sequence` to `validation.test.js` because it feels related to the validation.

Reply:
Great idea, thanks. Please rebase this PR as #5632 is now merged.